tls/s2n_kem.c

```diff
@@ -232,7 +232,7 @@ S2N_RESULT s2n_kem_generate_keypair(struct s2n_kem_params *kem_params)
     ENSURE(kem_params->public_key.size == kem->public_key_length, S2N_ERR_SAFETY);
 
     /* Need to save the private key for decapsulation */
-    GUARD_AS_RESULT(s2n_alloc(&kem_params->private_key, kem->private_key_length));
+    GUARD_AS_RESULT(s2n_realloc(&kem_params->private_key, kem->private_key_length));
```

Comment:
Basically. Previously, we only generated the private key once, so "alloc" (which assumes the blob has never been allocated before) was fine. Now, we might regenerate it in response to a HRR, so we have to be able to reuse the allocated memory, which is what "realloc" does.

I wonder if there's ever a reason use s2n_alloc instead of s2n_realloc? It kind of seems like a foot gun.

Reply:
@camshaft is all over this: https://github.com/aws/s2n-tls/pull/2641/files